src/wallet/rpcdump.cpp

```diff
@@ -557,7 +553,7 @@ UniValue importwallet(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_WALLET_ERROR, "Wallet is currently rescanning. Abort existing rescan or wait.");
     }
 
-    int64_t nTimeBegin = 0;
+    int64_t nTimeBegin = TIMESTAMP_MIN;
```

Comment:
This is a dead store since `nTimeBegin` is unconditionally written to on L573.

Perhaps you could drop the dead store and instead move the `nTimeBegin = chainActive.Tip()->GetBlockTime();` to the line after `LOCK(pwallet->cs_wallet);`?

Reply:
`nTimeBegin` is used outside that block for the rescan, line 646.
I prefer to leave this as it seems harmless. Alternative is leaving it uninitialized.

Reply:
What I meant is that the current code is:

```
int64_t nTimeBegin = TIMESTAMP_MIN;
nTimeBegin = chainActive.Tip()->GetBlockTime();
RescanWallet(*pwallet, reserver, nTimeBegin, false /* update */);
```

Thus the first assignment is redundant since it is overwritten in all possible execution paths.

There are no execution paths where dropping the first assignment would result in an uninitialised variable being read.